### PR TITLE
Datastore: addNullBinding and setNullBinding

### PR DESCRIPTION
**Pull request by andreamlin**
Resolves https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1559.

### CODE REVIEW

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/SerializationTest.java [hunk 1]
```diff
@@ -52,6 +52,12 @@
       .addBinding(20)
       .setNamespace("ns1")
       .build();
+  private static final Query<Entity> GQL3 =
+      Query.newGqlQueryBuilder(
+          Query.ResultType.ENTITY, "select * from kind1")
```
Comment (garrettjonesgoogle) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/SerializationTest.java]: It doesn't seem like this query would make use of the null binding.

**Comment by garrettjonesgoogle** [comment]
@andreamlin do you want to add tests for this PR, or do you want to abandon it?

**Comment by andreamlin** [comment]
I'll add tests!

**Comment by andreamlin** [comment]
Added test.

**Pull request closed without merging**